nautilus/include/nautilus/std/ostream.h

```diff
@@ -122,7 +122,16 @@ class val<std::basic_ostream<CharT, Traits>> {
 	template <class T>
 	static void pipe(std::basic_ostream<CharT, Traits>* ptr, T value) {
 		auto& s = *ptr;
-		s << value;
+		if constexpr (std::is_pointer_v<T> && !std::is_same_v<std::remove_cv_t<T>, char*>) {
+			if constexpr (not std::is_const_v<std::remove_pointer_t<T>> or
+			              not std::is_same_v<std::remove_cv_t<std::remove_pointer_t<T>>, char>) {
+				s << std::showbase << std::hex << reinterpret_cast<const void*>(value);
+			} else {
+				s << value;
+			}
+		} else {
+			s << value;
```

Comment:
```suggestion
		if constexpr (std::is_pointer_v<T> && !std::is_same_v<std::remove_cv_t<T>, char*> && ( !std::is_const_v<std::remove_pointer_t<T>> ||
			              !std::is_same_v<std::remove_cv_t<std::remove_pointer_t<T>>, char>) {
				s << std::showbase << std::hex << reinterpret_cast<const void*>(value);
			
		} else {
			s << value;
```

I would prefer to keep using "!" "&&" and "||" in the nautilus code base just to have the code consistant. 

Also we can remove a lavel of nesting